### PR TITLE
Fix a few more c++ build errors (rodete)

### PR DESCRIPTION
**Pull request by rsgowman**
* Fix implicit conversions from ints to chars.

These conversions can alter the values, so the rodete compiler objects
with messages like:
```
error: conversion to ‘char’ from ‘int’ may alter its value
error: conversion to ‘char’ alters ‘int’ constant value
```
The second one in particular is slightly alarming, and results from
things like this: `char c = 0xff;`

However, this is contained within tests dealing with odd corner cases,
so I've just cast'd the problem away.  (i.e. `static_cast<char>(0xff)`)

* Add absl_strings to firebase_firestore_util_test dependencies

Required to link the test on rodete.

### CODE REVIEW

File: Firestore/core/test/firebase/firestore/util/ordered_code_test.cc [hunk 1]
```diff
@@ -359,12 +359,13 @@ TEST(OrderedCodeInvalidEncodingsTest, NonCanonical) {
   for (int n = 2; n <= 10; ++n) {
     // Header with 1 sign bit and n-1 size bits.
     std::string header =
-        std::string(n / 8, 0xff) + std::string(1, 0xff << (8 - (n % 8)));
+        std::string(n / 8, static_cast<char>(0xff)) +
```
Comment (zxu123) [inline Firestore/core/test/firebase/firestore/util/ordered_code_test.cc]: or simpler form '\xff'

Comment (rsgowman) [reply Firestore/core/test/firebase/firestore/util/ordered_code_test.cc]: Done.

**Comment by wilhuff** [comment]
I just pushed a change that removed the -Wconversion flag. These are overstrict and annoying without really adding a ton of value.

**Comment by rsgowman** [comment]
Ok; Gil's changes largely make this entire PR irrelevant. I'll abandon this one and create a separate one for the remaining issue.
